File: CRUD/Java/app/src/main/java/com/fj/tpamobile/crud_java/activities/AddNewsActivity.java
package com.fj.tpamobile.crud_java.activities;

import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Toast;

import com.fj.tpamobile.crud_java.R;
import com.fj.tpamobile.crud_java.data.Utils;
import com.fj.tpamobile.crud_java.models.News;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

public class AddNewsActivity extends AppCompatActivity implements View.OnClickListener {

    private String getId;

    private EditText edtTitle;
    private EditText edtContent;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_news);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        FloatingActionButton fab = findViewById(R.id.fab);
        ImageView imgDelete = findViewById(R.id.img_delete);
        edtTitle = findViewById(R.id.edt_title);
        edtContent = findViewById(R.id.edt_content);

        if (getIntent().hasExtra(Utils.NewsId)){
            imgDelete.setVisibility(View.VISIBLE);

            getId = getIntent().getStringExtra(Utils.NewsId);
            DatabaseReference dbRef = FirebaseDatabase
                    .getInstance()
                    .getReference("news")
                    .child(getId);

            dbRef.addListenerForSingleValueEvent(new ValueEventListener() {
                @Override
                public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                    News news = dataSnapshot.getValue(News.class);

                    edtTitle.setText(news.getTitle());
                    edtContent.setText(news.getContent());
                }

                @Override
                public void onCancelled(@NonNull DatabaseError databaseError) {
                    Log.d("InfoAddNews", databaseError.getMessage());
                }
            });
        }

        fab.setOnClickListener(this);
        imgDelete.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()){
            case R.id.fab:
                String title = edtTitle.getText().toString();
                String content = edtContent.getText().toString();

                if (title.isEmpty()){
                    edtTitle.setError("Must be filled");
                }else if (content.isEmpty()){
                    edtContent.setError("Must be filled");
                }else{
                    DatabaseReference dbRef = FirebaseDatabase
                            .getInstance()
                            .getReference("news");

                    if (getIntent().hasExtra(Utils.NewsId)){
                        dbRef.child(getId).child("title").setValue(title);
                        dbRef.child(getId).child("content").setValue(content);
                    }else{
                        DatabaseReference dbAdd = dbRef.push();
                        dbAdd.child("title").setValue(title);
                        dbAdd.child("content").setValue(content);
                    }

                    Toast.makeText(this, "Successfully", Toast.LENGTH_SHORT).show();
                    finish();
                }
                break;
            case R.id.img_delete:
                new AlertDialog.Builder(this)
                        .setTitle("Are you sure?")
                        .setMessage("Delete this news?")
                        .setNegativeButton(android.R.string.no, null)
                        .setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
                            @Override
                            public void onClick(DialogInterface dialog, int which) {
                                FirebaseDatabase
                                        .getInstance()
                                        .getReference("news")
                                        .child(getId)
                                        .removeValue();

                                Toast.makeText(AddNewsActivity.this, "Successfully", Toast.LENGTH_SHORT).show();
                                finish();
                            }
                        })
                    .show();
                break;
        }
    }
}

File: CRUD/Kotlin/app/src/main/java/com/fj/tpamobile/crud_kotlin/activities/AddNewsActivity.kt
package com.fj.tpamobile.crud_kotlin.activities

import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.support.v7.app.AppCompatActivity
import android.view.View
import com.fj.tpamobile.crud_kotlin.R
import com.fj.tpamobile.crud_kotlin.data.Utils
import com.fj.tpamobile.crud_kotlin.models.News
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener

import kotlinx.android.synthetic.main.activity_add_news.*
import kotlinx.android.synthetic.main.content_add_news.*
import org.jetbrains.anko.toast

class AddNewsActivity : AppCompatActivity(), View.OnClickListener {

    private lateinit var getId: String

    override fun onClick(v: View?) {
        when(v){
            fab -> {
                val title = edt_title.text.toString()
                val content = edt_content.text.toString()

                when{
                    title.isEmpty() -> edt_title.error = "Must be filled"
                    content.isEmpty() -> edt_content.error = "Must be filled"
                    else -> {
                        val dbRef = FirebaseDatabase
                            .getInstance()
                            .getReference("news")

                        if (intent.hasExtra(Utils.NewsId)){
                            dbRef.child(getId).child("title").setValue(title)
                            dbRef.child(getId).child("content").setValue(content)
                        }else{
                            val dbAdd = dbRef.push()
                            dbAdd.child("title").setValue(title)
                            dbAdd.child("content").setValue(content)
                        }

                        toast("Successfully!")
                        finish()
                    }
                }
            }

            img_delete -> {
                AlertDialog.Builder(this)
                    .setTitle("Are you sure?")
                    .setMessage("Delete this news?")
                    .setNegativeButton(android.R.string.no, null)
                    .setPositiveButton(android.R.string.yes) { _, _ ->
                        FirebaseDatabase
                            .getInstance()
                            .getReference("news")
                            .child(getId)
                            .removeValue()

                        toast("Successfully")
                        finish()
                    }
                    .show()
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_news)
        setSupportActionBar(toolbar)

        if (intent.hasExtra(Utils.NewsId)){
            img_delete.visibility = View.VISIBLE

            getId = intent.getStringExtra(Utils.NewsId)
            val dbRef = FirebaseDatabase
                .getInstance()
                .getReference("news")
                .child(getId)

            dbRef.addListenerForSingleValueEvent(object : ValueEventListener {
                override fun onCancelled(p0: DatabaseError) {
                    println("InfoAddNews: ${p0.message}")
                }

                override fun onDataChange(p0: DataSnapshot) {
                    val getNews = p0.getValue(News::class.java)

                    edt_title.setText(getNews?.title)
                    edt_content.setText(getNews?.content)
                }
            })
        }

        fab.setOnClickListener(this)
        img_delete.setOnClickListener(this)
    }
}
